src/main/java/com/google/edith/servlets/UserInsights.java

```diff
@@ -33,61 +35,64 @@
 
   private String userId;
 
-  private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();  
-  private static final DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+  private static final DatastoreService datastore = 
+      DatastoreServiceFactory.getDatastoreService();  
+  private static final DateTimeFormatter dateFormatter = 
+      DateTimeFormatter.ofPattern("yyyy-MM-dd");
   private static final Gson gson = new Gson();
 
-  /** Assumes dates are in the form yyyy-MM-dd */
-  private static final Comparator<Key> SORT_BY_DATE = (Key item1, Key item2) -> {
-      try {
-        return ((String) (datastore.get(item1).getProperty("date")))
-            .compareTo((String) (datastore.get(item2).getProperty("date")));
-      } catch(EntityNotFoundException | NullPointerException e )  {
-        return 0;
-      }
-  };
-  
   public UserInsights(String userId) {
```

Comment:
This class seems almost like a 'UserInsightsService" rather than "UserInsights". When I think of "UserInsights", I think of a container class that stores relevant insights info for a particular user, however this class seems to be doing some fetching/processing w.r.t. user insights.

Thinking about this class from a service perspective, it makes a bit more sense to keep the datastore initialized via the constructor and pass the userIds in methods so that this service can be invoked for any user. 

If this class does truly _need_ to be initialized for every user (i.e. create a new UserInsights) that we need insights, I worry there might be a deeper dependency between a "user id" and "insights". And ideally they should be de-coupled or at least be able to be de-coupled.

Wdyt? Apologies for readdressing this, you might just need to refresh my memory. Lmk in case I misunderstood anything

Reply:
I understand you. I made a couple to changes that I think defines it more as a service.